IPython/core/magics/basic.py

```diff
@@ -294,13 +294,9 @@ def profile(self, parameter_s=''):
         prun : run code using the Python profiler
                (:meth:`~IPython.core.magics.execution.ExecutionMagics.prun`)
         """
-        warn("The `%profile` magic has been deprecated since IPython 2.0. Please use "
-             "the value of `get_ipython().profile` instead to see current profile in use.")
-        from IPython.core.application import BaseIPythonApplication
-        if BaseIPythonApplication.initialized():
-            print(BaseIPythonApplication.instance().profile)
-        else:
-            error("profile is an application-level value, but you don't appear to be in an IPython application")
+        raise DeprecationWarning("The `%profile` magic has been deprecated since IPython 2.0. "
```

Comment:
No, I meant raise. It has been printing a warning for already 4 versions.
The alternative is to just remove it, but then there is no explanation message. 
I want at least 1 version where calling this function is an error (raise exception or not present). To not have 2 consecutive versions where `%profile` is a valid magic.